### ISSUE TITLE
Additional column in the email display for marking messages with notes

### ISSUE THREAD

**Issue by RicoAlexanderSchulz** [xnotepp/xnote#136]
Additional column in the email display for marking messages with notes

There is a problem that the position appears to be incorrect, meaning that xNote inserted after the last column always produces a side scroll bar.

Hence my question
Could xNote be expanded so that I can switch off the XNote column so that I no longer have a side scroll bar?
Or to improve the programming so that the side scroll bar is prevented.

(All columns are narrow and there should be no side scroll bar.)
(If I disable xNote completely, the scroll bar is gone.)

Thank you

.
![2024-05-15_082003](https://github.com/xnotepp/xnote/assets/136677520/7dac3dc5-d5f9-41f6-bcd5-4c8f5baa0082)

.

.

German:
Zusätzliche Spalte in der E-Mail Anzeige zur Markierung von Nachrichten mit Notizen
Es besteht das Problem, das scheinbar die Position falsch ist, heißt, das xNote nach der letzten Spalte eingesetzt wird, erzeugt immer ein seitlichen Scrollbalken.

Daher meine Frage
Könnte man xNote so erweitern, das man die Spalte XNote abschalten kann, damit ich kein seitlichen Scrollbalken mehr habe?
Oder die Programmierung so zu verbessern, das der seitliche Scrollbalken verhindert wird.

(Alle Spalten sind schmal und es sollte kein seitlichen Scrollbalken geben.)
(Wenn ich xNote komplett deaktiviere, ist der seitliche Scrollbalken weg.)

Vielen Dank

**Comment by opto** [comment]
finally, after four years, the TB addon team has reintroduced "custom columns". That feature got lost somewhere during Thunderbird code rewriting, and I have been asking for that since then.

Currently, I am using a wild hack to write the XNote into a regular column. It took really long to find how to do that. So at least we are having the column back since about half a year, even if it is a bit ugly.

I will convert that to the new custom columns provided by Thnderbird in the near future.

**Comment by opto** [comment]
you can pull the column to a different position (more to the left) and use the column picker to set that for all folders. Does that help with the scrollbar? Or maybe deselct columns you don't need?

**Comment by RicoAlexanderSchulz** [comment]
The xNote column cannot be moved, it seems to be linked to the SpamStatus column, because when I take the xNote icon and move it to the desired position, the spam status column is always moved instead, even if it is 3 columns further away.

Noticeably, there are two menu logos and the latter cannot be used.

Windows 10
Thunderbird 115

Maybe switching it off will also help, which is currently not included in the menu.

Thank you for your valuable time and effort
It works so far and that's the most important thing.

![2024-05-15_110834](https://github.com/xnotepp/xnote/assets/136677520/76f77f03-00ca-4aa9-9349-72e317174cf2)

**Comment by reinerblock** [comment]
Just tested it because I'm missing the column for XNote++ and QNote. I've both installed, and my test shows different results.

If I create a XNote via context menu, the following is shown.
![grafik](https://github.com/xnotepp/xnote/assets/5745579/864d4de4-7263-4b39-8892-792c38af3828)

As you can see, there's a tag XNote now above the message, but the preview shows QNote.

Furthermore, the XNote column is not linked to the Spam or Status column but the Size column.

BTW, I'm using Betterbird 115.11.0-bb28.

**Comment by yardstudio** [comment]
@opto  how is this resolved?

**Comment by deakjahn** [comment]
@opto No, it isn't resolved. There are a few issues here (like https://github.com/xnotepp/xnote/issues/141) that are seemingly the same issue. Despite the latest upgrades for several months now, absolutely nothing has changed here.

All folders, even those where XNote++ is not used at all, appear with an extra column that throws off the horizontal scrolling. In situations where there's no need for scrolling at all because the information displayed fits into the window, this extra column forces the scrollbar to appear. It's not only unnecessary but it actively covers part of the last line and makes any manipulation of the last message very cumbersome.

You should address the issue finally.

**Comment by opto** [comment]
Interesting. Then, when I added resolved, the column clearly was not displayed. So something changed again in TB.
So I need to look into it again.

**Comment by deakjahn** [comment]
This is my current look of a folder without using XNote++.

![Image](https://github.com/user-attachments/assets/82c63dca-7180-4e26-8720-cd18b82b1189)

TB 128.7.0esr (64-bit), it says up-to-date as of right now.

**Comment by opto** [comment]
actually, this is resolved now, and will come automatically, but review is slow at the moment.

**Comment by RicoAlexanderSchulz** [comment]
Ok, Thank Youe
:-)

Am Do., 27. Feb. 2025 um 07:54 Uhr schrieb Klaus Buecher <
***@***.***>:
